include/kodlab_mjbots_sdk/mjbots_robot_interface.h

```diff
@@ -118,20 +102,18 @@ class MjbotsRobotInterface {
    * @brief accessor for the joint modes
    * @return the joint modes
    */
-  std::vector<::mjbots::moteus::Mode> GetJointModes();
+  std::vector< ::mjbots::moteus::Mode>  GetJointModes();
 
  private:
   int num_servos_;                         /// The number of motors in the robot
-  std::vector<int> servo_id_list_;         /// Vector of the servo id
-  std::vector<int> servo_bus_list_;        /// Vector of the servo bus
   std::map<int, int> servo_bus_map_;       /// map from servo id to servo bus
-  std::vector<float> positions_;           /// Vector of the motor positions
-  std::vector<float> velocities_;          /// Vector of the motor velocities
-  std::vector<float> torque_cmd_;          /// Vector of the torque command sent to motors
-  std::vector<float> offsets_;             /// Offset of the motor position
-  std::vector<int> directions_;            /// Direction of motors
+  std::vector<std::reference_wrapper<const float>> positions_;  /// Vector of the motor positions
+  std::vector<std::reference_wrapper<const float>> velocities_; /// Vector of the motor velocities
+  std::vector<std::reference_wrapper<const float>> torque_cmd_; /// Vector of the torque command sent to motors
+  std::vector<std::reference_wrapper<const ::mjbots::moteus::Mode>> modes_; /// Vector of the torque command sent to motors
+  
   std::shared_ptr<bool> timeout_ = std::make_shared<bool>(false);                   /// True if communication has timed out
-  std::vector<::mjbots::moteus::Mode> modes_;/// Vector of the motor modes
+  std::vector<JointMoteus> joints_;
```

Comment:
Lets put this before the  positions, velocities, and torques in the .h and add comments explaining that references and memory sharing.